py-polars/polars/dataframe/frame.py

```diff
@@ -1609,6 +1611,203 @@ def raise_on_copy(msg: str) -> None:
 
         return out
 
+    @overload
+    def to_jax(
+        self,
+        return_type: Literal["array"] = ...,
+        *,
+        device: jax.Device | str | None = ...,
+        label: str | Expr | Sequence[str | Expr] | None = ...,
+        features: str | Expr | Sequence[str | Expr] | None = ...,
+        dtype: PolarsDataType | None = ...,
+        order: IndexOrder = ...,
+    ) -> jax.Array: ...
+
+    @overload
+    def to_jax(
+        self,
+        return_type: Literal["dict"],
+        *,
+        device: jax.Device | str | None = ...,
+        label: str | Expr | Sequence[str | Expr] | None = ...,
+        features: str | Expr | Sequence[str | Expr] | None = ...,
+        dtype: PolarsDataType | None = ...,
+        order: IndexOrder = ...,
+    ) -> dict[str, jax.Array]: ...
+
+    def to_jax(
+        self,
+        return_type: JaxExportType = "array",
```

Comment:
I'd generally prefer to have a boolean flag for options that only have two possible values - not sure if you intend to add more return types in the future? Otherwise, an `as_dict` boolean would possibly be cleaner. But then again, `to_torch` has the same API and has three options... probably fine to keep it this way.

Reply:
Indeed, there may be a need for other options in the future, so it's a bit of forward-planning - and it has a clean correspondence with the equivalent `to_torch` API 👌